skins/oasis/modules/FollowedPagesController.class.php

```diff
@@ -3,37 +3,40 @@ class FollowedPagesController extends WikiaController {
 
 	const MAX_FOLLOWED_PAGES = 6;
 
-	public function executeIndex($params) {
-	
-		$page_owner = User::newFromName($this->wg->Title->getText());
+	public function index() {
+		$pageOwner = User::newFromName( $this->wg->Title->getText() );
 
-		if ( !is_object( $page_owner ) || $page_owner->getId() == 0 ) {
+		if ( !is_object( $pageOwner ) || $pageOwner->getId() == 0 ) {
 			// do not show module if page owner does not exist or is an anonymous user
 			return false;
 		}
 
 		// add CSS for this module
-		$this->wg->Out->addStyle(AssetsManager::getInstance()->getSassCommonURL("skins/oasis/css/modules/FollowedPages.scss"));
+		$this->wg->Out->addStyle( AssetsManager::getInstance()->getSassCommonURL( "skins/oasis/css/modules/FollowedPages.scss" ) );
 
-		$showDeletedPages = isset($params['showDeletedPages']) ? (bool) $params['showDeletedPages'] : true;
+		$showDeletedPages = $this->request->getBool( 'showDeletedPages', true );
 		// get 6 followed pages
-		$watchlist = FollowModel::getWatchList( $page_owner->getId(), 0, 6, null, $showDeletedPages );
-		$data = array();
+		$watchlist = FollowModel::getWatchList( $pageOwner->getId(), 0, 6, null, $showDeletedPages );
+		$data = [ ];
 		// weird.  why is this an array of one element?
-		foreach ($watchlist as $unused_id => $item) {
+		foreach ( $watchlist as $unused_id => $item ) {
 			$pagelist = $item['data'];
-			foreach ($pagelist as $page) {
+			foreach ( $pagelist as $page ) {
 				$data[] = $page;
 			}
 
 		}
 		// only display  your own page
-		if ($page_owner->getId() == $this->wg->User->getId()) {
-			$this->follow_all_link = Wikia::specialPageLink('Following', 'oasis-wikiafollowedpages-special-seeall', 'more');
+		if ( $pageOwner->getId() == $this->wg->User->getId() ) {
+			$this->follow_all_link = Linker::linkKnown(
+				SpecialPage::getTitleFor( 'Following' ),
+				wfMessage( 'oasis-wikiafollowedpages-special-seeall' )->escaped(),
+				[ 'class' => 'more' ]
+			);
 		}
-		
+
 		$this->data = $data;
-		$this->max_followed_pages = min(self::MAX_FOLLOWED_PAGES, count($this->data));
-		
+		$this->max_followed_pages = min( self::MAX_FOLLOWED_PAGES, count( $this->data ) );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)